src/sentry/receivers/onboarding.py

```diff
@@ -388,6 +362,32 @@ def record_user_context_received(project, event, **kwargs):
 
 
 event_processed.connect(record_user_context_received, weak=False)
+
+
+@first_event_with_minified_stack_trace_received.connect(weak=False)
+def record_event_with_first_minified_stack_trace_for_project(project, event, **kwargs):
+    try:
+        user: APIUser = Organization.objects.get(id=project.organization_id).get_default_owner()
+    except IndexError:
+        logging.getLogger("sentry").warning(
+            "Cannot record first event for organization (%s) due to missing owners",
+            project.organization_id,
+        )
+        return
+
+    if not project.flags.has_minified_stack_trace:
+        analytics.record(
+            "first_event_with_minified_stack_trace_for_project.sent",
+            user_id=user.id,
+            organization_id=project.organization_id,
+            project_id=project.id,
+            platform=event.platform,
+            url=dict(event.tags).get("url", None),
+        )
+
+    project.update(flags=F("flags").bitor(Project.flags.has_minified_stack_trace))
```

Comment:
Seems a bit complicated.
It looks like you update the project flag every time.
We are checking if we have the flag set in the previous statement so we can update in the if only once
(i.e. do the project update inside the if).